packtools/sps/validation/footnotes.py

```diff
@@ -80,67 +81,171 @@ def fn_validation(self):
+                    sub_item="fn",
+                    validation_type="match",
+                    is_valid=False,
+                    expected='<fn fn-type="conflict">',
+                    obtained='<fn fn-type="coi-statement">',
+                    advice="replace coi-statement with conflict",
+                    data=self.fns_dict,
+                    error_level=error_level
+                )
+
+    def missing_corresp_label_validation(self, error_level="WARNING"):
+        if bool(self.fns_dict.get("corresp")) and not bool(self.fns_dict.get("corresp_label")):
+            yield format_response(
+                title="Missing corresp label validation",
+                parent=self.fns_dict.get("parent"),
+                parent_id=self.fns_dict.get("parent_id"),
+                parent_article_type=self.fns_dict.get("parent_article_type"),
+                parent_lang=self.fns_dict.get("parent_lang"),
+                item="corresp",
+                sub_item="label",
+                validation_type="exist",
+                is_valid=False,
+                expected="<label> in <corresp>",
+                obtained=self.fns_dict.get("corresp_label"),
+                advice="Provide <label> in <corresp>",
+                data=self.fns_dict,
+                error_level=error_level
+            )
+
+    def missing_fn_label_validation(self, error_level="WARNING"):
+        for fn in self.fns_dict.get("fns"):
+            if not bool(fn.get("fn_label")):
+                yield format_response(
+                    title="Missing fn label validation",
+                    parent=self.fns_dict.get("parent"),
+                    parent_id=self.fns_dict.get("parent_id"),
+                    parent_article_type=self.fns_dict.get("parent_article_type"),
+                    parent_lang=self.fns_dict.get("parent_lang"),
+                    item="fn",
+                    sub_item="label",
+                    validation_type="exist",
+                    is_valid=False,
+                    expected="<label> in <fn>",
+                    obtained=fn.get("fn_label"),
+                    advice="Provide <label> in <fn>",
+                    data=self.fns_dict,
+                    error_level=error_level
+                )
+
+    def title_presence_in_corresp_validation(self, error_level="ERROR"):
+        if bool(self.fns_dict.get("corresp")) and bool(self.fns_dict.get("corresp_title")):
+            yield format_response(
+                title="Title presence in corresp validation",
+                parent=self.fns_dict.get("parent"),
+                parent_id=self.fns_dict.get("parent_id"),
+                parent_article_type=self.fns_dict.get("parent_article_type"),
+                parent_lang=self.fns_dict.get("parent_lang"),
+                item="corresp",
+                sub_item="title",
+                validation_type="exist",
+                is_valid=False,
+                expected="<title> not in <corresp>",
```

Comment:
@Rossi-Luciano Trocar por `expected="<label> in <corresp>"`

Reply:
OK, @robertatakenaka